src/domains/boundingbox.jl

```diff
@@ -3,52 +3,54 @@ boundingbox(d::Vector{T}) where {T <: Number} = minimum(d)..maximum(d)
 boundingbox(d::Set{T}) where {T<:Number} = minimum(d)..maximum(d)
 
 "Return the bounding box of the union of two or more bounding boxes."
-unionbox(d::Domain) = d
-unionbox(d1::Domain, d2::Domain) = unionbox(promote_domains(d1, d2)...)
-unionbox(d1::Domain, d2::Domain, domains...) =
+unionbox(d) = d
+unionbox(d1, d2) = unionbox1(promote_domains(d1, d2)...)
+unionbox(d1, d2, domains...) =
     unionbox(unionbox(d1,d2), domains...)
-
-unionbox(d1::Domain{T}, d2::Domain{T}) where {T} = unionbox1(d1, d2)
 unionbox1(d1, d2) = unionbox2(d1, d2)
 unionbox2(d1, d2) = fullspace(d1)
 unionbox1(d1::EmptySpace, d2) = d2
 unionbox1(d1::FullSpace, d2) = d1
 unionbox2(d1, d2::EmptySpace) = d1
 unionbox2(d1, d2::FullSpace) = d2
 
-unionbox(d1::D, d2::D) where {D<:FixedInterval} = d1
+unionbox1(d1::D, d2::D) where {D<:FixedInterval} = d1
```

Comment:
What's `unionbox1` and below `intersectbox1`?

Reply:
`unionbox(d1,d2)` is a function of two arguments, which sometimes simplifies for certain `d1` (independently of `d2` or for some possibilities of `d2`). Similarly, it may simplify for certain `d2`. The call chain goes:
`unionbox(d1,d2)` -> `unionbox1(d1,d2)` -> `unionbox2(d1,d2)` -> some_default_algorithm

The idea is that unionbox can dispatch safely (without ambiguity) solely on d1, and unionbox2 can dispatch on d2. I've used this pattern in many places and it works rather well. If a combination of `d1` and `d2` is specific enough, then one can dispatch on it using just the `unionbox` function.